src/features/applicationMetadata/ApplicationMetadataProvider.tsx

```diff
@@ -2,68 +2,80 @@ import React, { useEffect } from 'react';
 import type { PropsWithChildren } from 'react';
 
 import { useQuery } from '@tanstack/react-query';
+import type { UseQueryOptions } from '@tanstack/react-query';
 
-import { useAppQueries } from 'src/core/contexts/AppQueriesProvider';
 import { delayedContext } from 'src/core/contexts/delayedContext';
 import { createQueryContext } from 'src/core/contexts/queryContext';
-import { OldVersionError } from 'src/features/applicationMetadata/OldVersionError';
+import { onEntryValuesThatHaveState } from 'src/features/applicationMetadata/appMetadataUtils';
+import { VersionErrorOrChildren } from 'src/features/applicationMetadata/VersionErrorOrChildren';
+import { fetchApplicationMetadata } from 'src/queries/queries';
+import { getInstanceIdRegExp } from 'src/utils/instanceIdRegExp';
 import { isAtLeastVersion } from 'src/utils/versionCompare';
+import type { ApplicationMetadata, IncomingApplicationMetadata } from 'src/features/applicationMetadata/types';
 
 export const MINIMUM_APPLICATION_VERSION = {
   build: '8.0.0.108',
   name: 'v8.0.0',
 };
 
 // Also used for prefetching @see appPrefetcher.ts
-export function useApplicationMetadataQueryDef() {
-  const { fetchApplicationMetadata } = useAppQueries();
+export function getApplicationMetadataQueryDef() {
   return {
     queryKey: ['fetchApplicationMetadata'],
     queryFn: fetchApplicationMetadata,
-  };
+    select: (data) => {
+      const onEntry = data.onEntry ?? { show: 'new-instance' };
+
+      return {
+        ...data,
+        isValidVersion:
+          !!data.altinnNugetVersion &&
+          isAtLeastVersion({
+            actualVersion: data.altinnNugetVersion,
+            minimumVersion: MINIMUM_APPLICATION_VERSION.build,
+          }),
+        onEntry,
+        isStatelessApp: isStatelessApp(onEntry.show),
```

Comment:
I'm not sure I'm fully on board here... Our 'is stateless app' thing has been easy to misinterpret for a long while now, since it doesn't mean "is this app stateless" but rather "are we stateless _now_". Since "being stateless" just means "we're currently showing a data task before we potentially proceed to initialize a new instance" it also means that when the user clicks on an `InstantiationButton` in a stateless task we immediately go from being stateless to no longer being stateless.

So, conceptually, is `isStatelessApp` a property of the "application metadata"? I would argue it's not, it relies on more than that - including the current URL/route (to see if we're in an instance or not). The way this used to work made a bit more sense to me, as it was a hook (even though it did not re-render your component when the route changed), but now I'm not quite sure the state will update when you go from stateless to stateful...? It does, at least, rely on when/how often this `select(data)` runs.